lib/customer-deposit-wallet/test/unit/Cardano/Wallet/Deposit/HTTP/JSON/JSONSpec.hs

```diff
@@ -0,0 +1,75 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE TypeApplications #-}
+
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Cardano.Wallet.Deposit.HTTP.JSON.JSONSpec
+    ( spec
+    ) where
+
+import Prelude
+
+import Cardano.Wallet.Deposit.HTTP.Types.JSON
+    ( Address
+    , ApiT (..)
+    , Customer
+    , CustomerList
+    )
+import Cardano.Wallet.Deposit.Read
+    ( fromRawAddress
+    , fromRawCustomer
+    )
+import Data.Aeson
+    ( FromJSON (..)
+    , ToJSON (..)
+    , decode
+    , encode
+    )
+import Test.Hspec
+    ( Spec
+    , describe
+    , it
+    )
+import Test.QuickCheck
+    ( Arbitrary (..)
+    , Property
+    , chooseInt
+    , property
+    , vectorOf
+    , (===)
+    )
+
+import qualified Data.ByteString as BS
+
+spec :: Spec
+spec =
+    describe "JSON serialization & deserialization" $ do
+        it "ApiT Address" $ property $
+            prop_jsonRoundtrip @(ApiT Address)
+        it "ApiT Customer" $ property $
+            prop_jsonRoundtrip @(ApiT Customer)
+        it "ApiT CustomerList" $ property $
+            prop_jsonRoundtrip @(ApiT CustomerList)
+
+prop_jsonRoundtrip :: (Eq a, Show a, FromJSON a, ToJSON a) => a -> Property
+prop_jsonRoundtrip val =
+    decode (encode val) === Just val
+
+instance Arbitrary (ApiT Address) where
+    arbitrary = do
+        --enterprise address type with key hash credential is 01100000, network (mainnet) is 1
+        --meaning first byte is 01100001 ie. 96+1=97
+        let firstByte = 97
+        keyhashCred <- BS.pack <$> vectorOf 28 arbitrary
+        pure $ ApiT $ fromRawAddress $ BS.append (BS.singleton firstByte) keyhashCred
+
+instance Arbitrary (ApiT Customer) where
+    arbitrary =
+        ApiT . fromRawCustomer <$> arbitrary
```

Comment:
```suggestion
        ApiT <$> chooseBoundedIntegral
```

🤓 On the matter of the `Word32 → Customer` conversion: Somewhat unusually, I have chose to make `Customer` a concrete type rather than an abstract data type — specifically, I have defined `Customer = Word31`. In other words, the deposit wallet will only support customer IDs in the range `0 … 2^31-1`!

This is intentional, and due to the BIP-32 address derivation, where the index for one derivation step is exactly a `Word31`. I could have used longer indices, but I decided to keep this limit for compatibility with hardware wallets; I don't want to explain derivations with arbitrary length. Hence, `Customer = Word31` is baked into the Deposit Wallet.

Reply:
introduced `fromRawCustomer` that creates `Customer` from `Word31` and added `Arbitrary Word31`.